components/core/src/os/users/admincheck.c

```diff
@@ -0,0 +1,90 @@
+#include <windows.h>
+#include <winnt.h>
+#include <ShlObj.h>
+
+#pragma comment (lib,"shell32.lib")
+
+TOKEN_ELEVATION_TYPE s_elevationType = TokenElevationTypeDefault;
+BOOL                 s_bIsAdmin = FALSE;
+
+// returns 0 if an admin
+// returns 1 if not
+int GetElevationState()
```

Comment:
If this is an int masquerading as a bool that we need for unsafe interface, consider changing name to be something like "IsElevatedAdmin" or some such.